Assets/LeapMotion/Core/Scripts/Attributes/SerializedTypeNameImplementsAttribute.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEditor;
+using UnityEngine;
+
+namespace Leap.Unity.Attributes {
+
+  /// <summary>
+  /// Place this attribute on a serialized string field to have it render as a dropdown
+  /// menu that is automatically populated with implementing types of the type provided
+  /// to the attribute. The string field will contain the type name selected by the user.
+  /// This can be converted into a Type reference via reflection and used to construct
+  /// objects by their type, for example, or to construct ScriptableObjects by their type.
+  /// </summary>
+  public class ImplementsTypeNameDropdownAttribute : CombinablePropertyAttribute,
+                                                     IFullPropertyDrawer {
+    
+    private Type _baseType;
+    private List<Type> _implementingTypes = new List<Type>();
+
+    public ImplementsTypeNameDropdownAttribute(Type type) {
+      _baseType = type;
+    }
+
+    public void DrawProperty(Rect rect, SerializedProperty property, GUIContent label) {
+      refreshImplementingTypes();
+
+      int curSelectedIdx = getCurSelectedIdx(property);
+
+      int selectedIdx = EditorGUI.Popup(rect, label, curSelectedIdx, getTypeOptions());
+      if (selectedIdx != curSelectedIdx) {
+        property.stringValue = _implementingTypes[selectedIdx].FullName;
+      }
+
+      if (curSelectedIdx == -1 && _implementingTypes.Count > 0) {
+        curSelectedIdx = 0;
+        property.stringValue = _implementingTypes[curSelectedIdx].FullName;
+      }
+    }
+
+    private void refreshImplementingTypes() {
```

Comment:
It would be really nice if this didn't search through all assemblies every time it was drawn, that could slow down the entire inspector if there are a few of these.  Since types cannot change without a recompile, you would be safe just caching the results.

Reply:
In fact you could populate both this list, and the list of gui content from within the constructor of this attribute!  No caching needed, just initialization!